apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/TraceHeaderPropagatingHandler.cs

```diff
@@ -24,73 +24,67 @@ namespace Google.Cloud.Diagnostics.Common
     /// <summary>
     /// Traces outgoing HTTP requests and propagates the trace header.
     /// </summary>
-    ///
+    /// 
     /// <example>
     /// <code>
     /// public void DoSomething(IManagedTracer tracer)
     /// {
-    ///     var traceHeaderHandler = TraceHeaderPropagatingHandler.Create(tracer);
+    ///     var traceHeaderHandler = new TraceHeaderPropagatingHandler(() => tracer);
     ///     using (var httpClient = new HttpClient(traceHeaderHandler))
     ///     {
     ///         ...
     ///     }
     /// }
     /// </code>
     /// </example>
+    ///
     /// <remarks>
     /// Ensures the trace header is propagated in the headers for outgoing HTTP requests and 
     /// traces the total time of the outgoing HTTP request.  This is only done if tracing is initialized
     /// and tracing is enabled for the request current request.
     /// </remarks>
     public class TraceHeaderPropagatingHandler : DelegatingHandler
     {
-        private readonly IManagedTracer _tracer;
+        private readonly Func<IManagedTracer> _managedTracerFactory;
 
-        private TraceHeaderPropagatingHandler(IManagedTracer tracer, HttpMessageHandler innerHandler)
+        internal TraceHeaderPropagatingHandler(
```

Comment:
If this is internal, the example won't work, right? I suspect we need it to be public.